### DISCUSSION TITLE
837 edi parsing

### DISCUSSION THREAD

**Discussion by cah-saviochandan-reddimasu** [nextgenhealthcare/connect#5478] [Q&A]
Hi....I'm looking to parse a bunch of raw 837's which I'm receiving in .txt format.
I'm able to set up the appropriate Data types in my source (type=file reader) (Inbound EDI-->Outboud EDI). That works in a sense that the encoded message puts it in a nice tag based structure by reading the source. Output type is file writer
Now I have an extra need.....To parse this message and convert it into csv format for the next team to consume....So I have a couple of questions....
1. Do I read it at the source to do some trickery, or 
2. create another task on destination to read the newly created structured XML file to do the conversion

If anyone can point me in the right direction with some code samples, I would greatly appreciate it.

**Comment by siddharthsharma9258** [comment]
You could use a source transformer.
There should be samples available already.

**Comment by jonbartels** [comment]
Check out https://forums.mirthproject.io/ . That is the old forums and has over a decade of old threads that might have good examples of X12 parsing in MC.
